evaluation/scripts/service_setup.py

```diff
@@ -0,0 +1,159 @@
+import logging
+import os
+
+import openai
+from azure.core.credentials import AzureKeyCredential
+from azure.identity import AzureDeveloperCliCredential, get_bearer_token_provider
+from azure.search.documents import SearchClient
+from promptflow.core import (
+    AzureOpenAIModelConfiguration,
+    ModelConfiguration,
+    OpenAIModelConfiguration,
+)
+from pyrit.chat_message_normalizer import ChatMessageNop, ChatMessageNormalizer
+from pyrit.prompt_target import (
+    AzureMLChatTarget,
+    AzureOpenAIChatTarget,
+    OpenAIChatTarget,
+    PromptChatTarget,
+)
+
+logger = logging.getLogger("scripts")
+
+def print_all_env_vars():
+    vars = ['OPENAI_HOST', 'OPENAI_GPT_MODEL', 'AZURE_OPENAI_EVAL_DEPLOYMENT', 'AZURE_OPENAI_SERVICE', 'OPENAICOM_KEY', 'OPENAICOM_ORGANIZATION', 'AZURE_SEARCH_SERVICE', 'AZURE_SEARCH_INDEX', 'AZURE_OPENAI_KEY', 'TENANT_ID', 'CLIENT_ID', 'CLIENT_SECRET', 'AZURE_PRINCIPAL_ID']
+    print('Environment Variables:')
+    for var in vars:
+        print(f"{var}: {os.environ.get(var)}")
+
+def get_openai_config() -> ModelConfiguration:
+    print_all_env_vars()
+    if os.environ.get("OPENAI_HOST") == "azure":
+        azure_endpoint = f"https://{os.environ['AZURE_OPENAI_SERVICE']}.openai.azure.com"
+        azure_deployment = os.environ.get("AZURE_OPENAI_EVAL_DEPLOYMENT")
+        api_version = "2023-07-01-preview"
+        if os.environ.get("AZURE_OPENAI_KEY"):
+            logger.info("Using Azure OpenAI Service with API Key from AZURE_OPENAI_KEY")
+            openai_config = AzureOpenAIModelConfiguration(
+                azure_endpoint=azure_endpoint,
+                azure_deployment=azure_deployment,
+                api_version=api_version,
+                api_key=os.environ["AZURE_OPENAI_KEY"],
+            )
+        else:
+            logger.info("Using Azure OpenAI Service with Azure Developer CLI Credential")
+            openai_config = AzureOpenAIModelConfiguration(
+                azure_endpoint=azure_endpoint,
+                azure_deployment=azure_deployment,
+                api_version=api_version,
+            )
+            # PromptFlow will call DefaultAzureCredential behind the scenes
+        openai_config.model = os.environ["OPENAI_GPT_MODEL"]
+    else:
+        logger.info("Using OpenAI Service with API Key from OPENAICOM_KEY")
+        openai_config = OpenAIModelConfiguration(
+            model=os.environ["OPENAI_GPT_MODEL"],
+            api_key=os.environ.get("AZURE_OPENAI_KEY"),
+            organization=os.environ["OPENAICOM_ORGANIZATION"],
+        )
```

Comment:
What does this correspond to? Does our application support it? Do we need it?

Reply:
I'm asking because this shows up in multiple places in the script and I'm afraid it only complicates things.

Reply:
It‘s for the gpt model on openai.com. It is optional and in our case, we basically not use it and I only used this one time when the gpt model was not available on Azure.